wcfsetup/install/files/lib/system/user/notification/event/TReactionUserNotificationEvent.class.php

```diff
@@ -0,0 +1,55 @@
+<?php
+namespace wcf\system\user\notification\event;
+use wcf\data\user\UserProfile;
+use wcf\system\user\notification\object\IUserNotificationObject;
+
+/**
+ * This trait can be used in likeable user notification events to determine the reactionType counts
+ * for a specific user notification object.
+ * 
+ * HEADS UP: This trait is only included in version 3.0 and 3.1 for compatibility reasons. This 
```

Comment:
I believe this backport is primarily to prevent using an non-existant trait (which will fail hard). As the methods are not usable anyway, consider either:

1. Removing the methods from the trait. They should not be used anywhere anyway.
2. Throwing an Exception whenever the methods are called to make it fail hard in case something erroneously relies on them.

Reply:
I wouldn't want to remove the methods. However, I think the idea of an exception when the method is used is good. 
Which exception is best in this case? The `BadMethodCallException`? `NotImplementedException` is unfortunately not compatible with version 3.0.

Reply:
`BadMethodCallException` is fine to me.